dlk/python/dlk/templates/src/matrix/shift_add.cpp

```diff
@@ -170,11 +171,10 @@ void matrix_shift_add(MatrixView<int32_t, MatrixOrder::ColMajor>& buf,
             }
           }
     }));
-  }
-
-  for (auto& th: threads) {
-    if (th.joinable())
-      th.join();
+    for (auto& th: threads) {
+      if (th.joinable())
+	th.join();
+    }
   }
```

Comment:
The change looks wrong, because exec `th.join()` in each `for (unsigned int begin = 0; begin < h * w; begin += chunk_size)` loop.

```
for (unsigned int begin = 0; begin < h * w; begin += chunk_size)

...
    for (auto& th: threads) {
      if (th.joinable())
	th.join();
    }
}
```

Is it OK ?